module/Finna/src/Finna/Controller/MyResearchController.php

```diff
@@ -1006,35 +1006,36 @@ public function changeMessagingSettingsAction()
         $config = $catalog->getConfig('updateMessagingSettings', $patron);
 
         if ($this->formWasSubmitted('messaging_update_request')) {
-            if (isset($config['method']) && 'driver' === $config['method']) {
-                $data = $profile['messagingServices'];
-                $request = $this->getRequest();
-                // Collect results from the POST request and update settings
-                foreach ($data as $serviceId => &$service) {
-                    foreach ($service['settings'] as $settingId => &$setting) {
-                        if (!empty($setting['readonly'])) {
-                            continue;
-                        }
-                        if ('boolean' == $setting['type']) {
-                            $setting['active'] = (bool)$request->getPost(
-                                $serviceId . '_' . $settingId,
+            $data = $profile['messagingServices'];
+            $request = $this->getRequest();
+            // Collect results from the POST request and update settings
+            foreach ($data as $serviceId => &$service) {
+                foreach ($service['settings'] as $settingId => &$setting) {
+                    if (!empty($setting['readonly'])) {
+                        continue;
+                    }
+                    if ('boolean' == $setting['type']) {
+                        $setting['active'] = (bool)$request->getPost(
+                            $serviceId . '_' . $settingId,
+                            false
+                        );
+                    } elseif ('select' == $setting['type']) {
+                        $setting['value'] = $request->getPost(
+                            $serviceId . '_' . $settingId,
+                            ''
+                        );
+                    } elseif ('multiselect' == $setting['type']) {
+                        foreach ($setting['options'] as $optionId => &$option) {
+                            $option['active'] = (bool)$request->getPost(
+                                $serviceId . '_' . $settingId . '_' . $optionId,
                                 false
                             );
-                        } elseif ('select' == $setting['type']) {
-                            $setting['value'] = $request->getPost(
-                                $serviceId . '_' . $settingId,
-                                ''
-                            );
-                        } elseif ('multiselect' == $setting['type']) {
-                            foreach ($setting['options'] as $optionId => &$option) {
-                                $option['active'] = (bool)$request->getPost(
-                                    $serviceId . '_' . $settingId . '_' . $optionId,
-                                    false
-                                );
-                            }
                         }
                     }
                 }
+            }
```

Comment:
Kun nyt tässä ollaan, niin luupin perään voisi laittaa unsetin, ettei jää viittaus roikkumaan:
```suggestion
            }
            unset($setting);
```